stdlib/src/builtin/string.mojo

```diff
@@ -1136,14 +1136,31 @@ struct String(
         elems.each[add_elt]()
         return result
 
-    fn _strref_dangerous(self) -> StringRef:
-        """
-        Returns an inner pointer to the string as a StringRef.
+    fn _strref_dangerous(self, start: Int = 0, length: Int = -1) -> StringRef:
```

Comment:
Please make these keyword-only to force callers to be explicit. Also, please use `Optional[Int]` for `length`:
```suggestion
    fn _strref_dangerous(self, *, start: Int = 0, length: Optional[Int] = None) -> StringRef:
```